crates/bevy_reflect/src/type_registry.rs

```diff
@@ -599,6 +602,39 @@ impl<T: Reflect> FromType<T> for ReflectFromPtr {
     }
 }
 
+/// Associate a `TraitTypeData`
+pub trait TraitTypeDataRelevance {
```

Comment:
Bikeshed: I think a name like `AsoociatedTypeData`, `TypeDataAssoc`, `TypeDataMapper`, etc. would be more indicative of how this is meant to be used.

Additionally, I think we can leave out the `Trait` part of the name since this doesn't necessarily need to be used for `dyn Trait`. For example, a user could associate a `Player` struct type with a `ReflectPlayerSettings` or something.